api/client.go

```diff
@@ -131,34 +131,26 @@ func (c *httpClient) Do(ctx context.Context, req *http.Request) (*http.Response,
 		req = req.WithContext(ctx)
 	}
 	resp, err := c.client.Do(req)
-	defer func() {
-		if resp != nil {
-			resp.Body.Close()
-		}
-	}()
-
 	if err != nil {
 		return nil, nil, err
 	}
 
 	var body []byte
-	done := make(chan struct{})
+	done := make(chan error, 1)
 	go func() {
 		var buf bytes.Buffer
-		_, err = buf.ReadFrom(resp.Body)
+		_, err := buf.ReadFrom(resp.Body)
 		body = buf.Bytes()
-		close(done)
+		done <- err
 	}()
 
 	select {
 	case <-ctx.Done():
+		resp.Body.Close()
```

Comment:
This assumes that close will somehow cancel `_, err := buf.ReadFrom(resp.Body)`. Is it true? How?

Can we have a unit tests that perhaps shows this? 🤔

Reply:
@bwplotka could you take a look 👀

Reply:
Sorry for lag! 

Yea, looks like this behaviour is documented also in HTTP docs:

```
	// Body must allow Read to be called concurrently with Close.
	// In particular, calling Close should unblock a Read waiting
	// for input. <--- this one.
	Body io.ReadCloser
```

It would be epic to have this test yes! However instead of 500ms delay, I would block it with` <-t.Context.Done()` or something. This will ensure we get `Do` returning without Go test timeout. 

Thanks!

Reply:
Resolved in new commit